README.md
# Persons
Manage persons. This is an extension for the TYPO3 CMS.

### Table of contents

* Configuration
    * [TypoScript](./Documentation/Configuration/TypoScript.md)
* Models
    * [Content](./Documentation/Models/Content.md)
    * [Person](./Documentation/Models/Content.md)
* Plugins
    * Person
### Credentials
Icons made by <a href="http://www.freepik.com" title="Freepik">Freepik</a> from <a href="http://www.flaticon.com" title="Flaticon">www.flaticon.com</a> is licensed by <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank">CC 3.0 BY</a></div>


ChangeLog
2019-09-30         [RELEASE] Release of persons 1.0.0 (Dirk Wenzel)
2019-09-30 eec0895 [TASK] template clean up (Dirk Wenzel)
2019-07-05 579f154 [WIP][FEATURE] Sort selected persons by given orderings from FlexForm refs #4 (Elias Häußler)
2019-07-05 c75a421 [TASK] Add default HTML list view for all persons (Elias Häußler)
2019-07-05 b6634a0 [BUGFIX] Ensure LLL keys in FlexForm can be found and interpreted refs #3 (Elias Häußler)
2019-07-04 0e31270 [TASK] Rewrite deprecated private attribute assertions in unit tests (Elias Häußler)
2019-07-04 2f6c98f [TASK] Remove legacy "_md5_values_when_last_written" from ext_emconf.php (Elias Häußler)
2019-07-04 9ea77e9 [!!!][TASK] Streamline line endings of TypoScript and TSconfig files (Elias Häußler)
2019-07-04 4979b67 [TASK] Update TCA of default fields for correct use within TYPO3 9.5 (Elias Häußler)
2019-07-03 25af826 [!!!][TASK] Add compatibility with TYPO3 9.5 and drop 8.7 compatibility (Elias Häußler)

2017-10-17         [RELEASE] Release of persons 0.6.1 (Dirk Wenzel)
2017-10-13 18d8a3f [BUGFIX] fixed error in PersonControllerSlot which prevented values of zero from being passed to configuration for frontend (Dirk Wenzel)
2017-10-12 31c7e64 [BUGFIX] Fetch only required property values for FileReferences (Nicole Cordes)

2017-10-11         [RELEASE] Release of persons 0.6.0 (Dirk Wenzel)
2017-10-11 f4ca8df [TASK] script tags and data attributes added to templates for filter and list view. (Dirk Wenzel)
2017-10-10 88a4157 [BUGFIX] fixed error where default language were not found in LocalizationUtility (Dirk Wenzel)
2017-10-10 cfa917f [TASK] PersonControllerSlot edited: pass keys for current language only (Dirk Wenzel)
2017-10-10 c4a3a07 [TASK] PersonControllerSlot now allows transferring additional keys in list action. LocalizationUtility extended by method getLanguageKey. Connect signals and slots. (Dirk Wenzel)
2017-10-10 ad38330 [TASK] LocalizationUtility and PersonControllerSlot implemented. (Dirk Wenzel)
2017-10-05 cb621fe [TASK] TypoScript documentation added. (Dirk Wenzel)
2017-10-05 65413b1 [TASK] obsolete settings removed from default TypoScript. Documentation added. (Dirk Wenzel)
2017-10-05 13d1d06 [BUGFIX] fixed error caused by missing method callStatic. CallStaticTrait implemented. (Dirk Wenzel)
2017-10-04 c3116e8 [TASK] ImageService implemented. AbstractJsonView extended: Now renders image URI for FileReference if configured in settings. Example for imageProcessing  added to TypoScript. Caching enabled in default configuration for personPage object. (Dirk Wenzel)

2017-09-29         [RELEASE] Release of persons 0.5.0 (Dirk Wenzel)
2017-09-29 62c00f5 [TASK] cast to bool replaced by !empty() in PersonRepository->findMultipleByUid (Dirk Wenzel)
2017-09-26 83b4352 [BUGFIX] Allow persons with default language in showSelectedAction (Nicole Cordes)

2017-09-26         [RELEASE] Release of persons 0.4.1 (Dirk Wenzel)
2017-09-26 46cf6d6 [BUGFIX] fixed error in AbstractJsonView caused by invalid return value null of TYPO3\CMS\Extbase\Persistence\Generic\LazyLoadingProxy->_loadRealInstance() (Dirk Wenzel)

2017-09-25         [RELEASE] Release of persons 0.4.0 (Dirk Wenzel)
2017-09-20 a2d564d [TASK] german localization for backend added. Changed language file format to xliff. (Dirk Wenzel)

2017-09-19         [RELEASE] Release of persons 0.3.0 (Dirk Wenzel)
2017-09-19 d8a8945 [TASK] german localization for backend added. (Dirk Wenzel)

2017-09-19         [RELEASE] Release of persons 0.2.0 (Dirk Wenzel)
2017-09-14 f0d1178 [BUGFIX] fixed error in configuration of json list view which prevented image field from rendering (Dirk Wenzel)
2017-09-14 6a6e2a6 [BUGFIX] View ListJson now renders missing field uid too. (Dirk Wenzel)
2017-09-13 fd34dd2 [TASK] PersonController->listAction now emits a signal before assigning variables to view too. (Dirk Wenzel)
2017-09-08 8d05a6a [TASK] Json view for person listing extended: now category parent is displayed too. (Dirk Wenzel)
2017-09-08 1218770 [TASK] PersonRepositoryTrait and SignalTrait implemented. PersonController refactored and extended: filterAction now sends signal before assigning variables to view. (Dirk Wenzel)
2017-08-23 cfa2cb0 [TASK] Json View now transforms object storages too. View ListJson renders categories. (Dirk Wenzel)
2017-08-03 03c8f8b [TASK] model Person edited: field status removed, departments replaced by categories (Dirk Wenzel)
2017-08-01 c8d4301 [BUGFIX] fix error in Person model where getter return wrong property. Failing unit test for PersonController fixed. (Dirk Wenzel)
2017-07-31 d6e76e7 [TASK] plugin now displays field pluginNamespace for each action (Dirk Wenzel)
2017-07-31 4d96828 [TASK] show pages configuration for all actions of person plugin (Dirk Wenzel)
2017-07-31 330337f [TASK] Variable assignment of filter action in PersonController changed. Json view configuration adapted. (Dirk Wenzel)
2017-07-31 46944e0 [TASK] Json view now transforms ObjectStorage objects to array (Dirk Wenzel)
2017-07-26 1ef7388 [TASK] Filter action for PersonController implemented (Dirk Wenzel)
2017-07-05 3b01902 [TASK] AbstractJsonView, Person/ListJson and Person/ShowSelectedJson of PersonController implemented. (Dirk Wenzel)
2017-07-04 322b083 [TASK] JsonView implemented (WIP) (Dirk Wenzel)
2017-07-03 49fa06e [TASK] missing property www added to Person. (Dirk Wenzel)

2017-06-29         [RELEASE] Release of persons 0.1.0 (Dirk Wenzel)
2017-06-29 631ee5c [BUGFIX] fixed error in composer.json which caused failing builds (Dirk Wenzel)
2017-06-22 7cccd20 [BUGFIX] invalid annotation in domain model Person fixed. (Dirk Wenzel)
2017-06-22 9ec26f1 [TASK] domain model Person extended by field departments. (Dirk Wenzel)
2017-06-21 22ec2cd [BUGFIX] TypoScript path corrected (Dirk Wenzel)
2017-06-16 6c44e6f [TASK] PersonRepository extended by methods findMultipleByUid and createOrderingsFromList. PersonController extended by showSelectedAction. Plugin can now display selected persons. (Dirk Wenzel)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "persons".
 *
 * Auto generated 17-10-2017 09:14
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

/** @noinspection PhpUndefinedVariableInspection */
$EM_CONF[$_EXTKEY] = [
    'title' => 'Persons',
    'description' => 'Manage persons',
    'category' => 'plugin',
    'author' => 'Dirk Wenzel',
    'author_email' => 'wenzel@cps-it.de',
    'state' => 'beta',
    'uploadfolder' => '1',
    'createDirs' => '',
    'clearCacheOnLoad' => 0,
    'version' => '1.0.0',
    'constraints' => [
        'depends' => [
            'typo3' => '9.5.0-9.5.99',
        ],
    ],
];

